### ISSUE TITLE
Pagination broken (at least for beta datastore emulator)

### ISSUE THREAD

**Issue by tianhuil** [googleapis/nodejs-datastore#500]
**TL;DR**: Recommended pagination termination check of `info.moreResults !== Datastore.NO_MORE_RESULTS` fails.  The variable `info.moreResults` is always`MORE_RESULTS_AFTER_LIMIT`, even if query returned no results.

## Steps to reproduce

  1. The code is available here: https://github.com/tianhuil/react-static-ts/blob/test-datastore/gae/backend/src/datastore.int.test.js
  2. The makefile is available here: https://github.com/tianhuil/react-static-ts/blob/test-datastore/gae/Makefile

Just clone the repo, cd into the GAE directory, run `npm i`, and `make test-datastore` after appropriate node.  You'll need a beta datastore emulator's host / port set to the environment variable `DATABASE_HOST`.

The code has two versions of the [pagination code](https://cloud.google.com/datastore/docs/concepts/queries#datastore-datastore-inequality-range-nodejs):
1. One checks only `info.moreResults !== Datastore.NO_MORE_RESULTS` and fails (infinite loop)
2. The other checks `info.moreResults !== Datastore.NO_MORE_RESULTS && entities.length > 0` and completes with the correct answer.

## Environment details

  - OS: OSX
  - Node.js version: v12.6.0
  - npm version: 6.11.3
  - `@google-cloud/datastore` version: 4.3.0

Thanks!

**Comment by stephenplusplus** [comment]
This is unfortunately a long standing bug with the emulator :( https://github.com/GoogleCloudPlatform/google-cloud-datastore/issues/130

**Comment by tianhuil** [comment]
Thanks!

**Comment by stephenplusplus** [comment]
Our library already does the NOT_FINISHED check. The user in an emulator could check for entities.length, although I’m not sure if that has any gotchas.
